lib/improver/spotdata/extract_data.py

```diff
@@ -0,0 +1,374 @@
+        # to a standard name if possible.
+        result_cube.rename(cube.name())
+        return result_cube
+
+    def use_nearest(self, cube, sites, neighbours):
+        """
+        Simplest case, in which the diagnostic data value at the nearest grid
+        point, as determined by the chosen PointSelection method, is used for
+        the site.
+
+        Args:
+        -----
+        See process() above.
+
+        """
+        if (not cube.coord_dims(cube.coord(axis='y').name())[0] == 0 or
+                not cube.coord_dims(cube.coord(axis='x').name())[0] == 1):
+            raise Exception("Cube dimensions not as expected.")
+
+        data = cube.data[neighbours['i'], neighbours['j']]
+        return self.make_cube(cube, data, sites)
+
+    def orography_derived_temperature_lapse_rate(self, cube, sites, neighbours,
+                                                 orography, no_neighbours=9):
+        """
+        Crude lapse rate method that uses temperature variation and height
+        variation across local nodes to derive lapse rate. This is highly
+        prone to noise given the small number of points involved and the
+        variable degree to which elevation changes across the small number
+        of points.
+
+        Args:
+        -----
+        cube : iris.cube.Cube
+            A cube of screen level temperatures at a single time.
+
+        sites/neighbours/no_neighbours : See process() above.
+
+        orography : numpy.array
+            Array of orography data on a grid that corresponds to the grid of
+            the the diagnostic cube.
+
+        Returns:
+        --------
+        Data extracted from the screen level temperature cube at spotdata sites
+        which has then been adjusted using a temperature lapse rate calculated
+        using local variations in temperature with orography.
+
+        """
+        def local_lapse_rate(cube, orography, node_list):
+            """
+            Least-squares fit to local temperature and altitude data for grid
+            points defined by node_list to calculate a local lapse rate.
+
+            """
+            y_data = cube.data[node_list]
+            x_data = orography[node_list]
+            matrix = np.vstack([x_data, np.ones(len(x_data))]).T
+            gradient, intercept = lstsq(matrix, y_data)[0]
+            return [gradient, intercept]
+
+        data = np.empty(shape=(len(sites)))
+
+        for i_site, site in enumerate(sites.itervalues()):
+            altitude = site['altitude']
+
+            i, j = neighbours['i'][i_site], neighbours['j'][i_site]
+            edgepoint = neighbours['edgepoint'][i_site]
+            node_list = nearest_n_neighbours(i, j, no_neighbours)
+            if edgepoint:
+                node_list = node_edge_test(node_list, cube)
+
+            llr = local_lapse_rate(cube, orography, node_list)
+            data[i_site] = llr[0]*altitude + llr[1]
```

Comment:
The local lapse rate has been constructed based on a relationship between temperature and orography, so `Temperature = gradient * Orography + y-intercept`. Here, `altitude` has been put into this least squares regression equation. Is there any significance for the change from `orography` to `altitude`?